src/Init/Data/Iterators/Combinators/Attach.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+module
+
+prelude
+import Init.Data.Iterators.Combinators.Monadic.Attach
+import Init.Data.Iterators.Combinators.FilterMap
+
+namespace Std.Iterators
+
+@[always_inline, inline, inherit_doc IterM.attachWith]
+def Iter.attachWith {α β : Type w}
+    [Iterator α Id β]
+    (it : Iter (α := α) β) (P : β → Prop) (h : ∀ out, it.IsPlausibleIndirectOutput out → P out) :=
```

Comment:
Maybe add the return type here? Makes it slightly easier to see what comes out.

Reply:
In this case it is possible to add the return type and I will, but for more complicated combinators, writing out the implicit parameter isn't super useful. That's why I have this pattern of returning `(something : Iter β)`. It lets me specify the kind of iterator that is returned without having to explicitly write down the implicit type parameter.

Reply:
I see, thanks for explaining.